community/kernel/src/main/java/org/neo4j/kernel/impl/storemigration/StoreUpgrader.java

```diff
@@ -116,8 +118,6 @@ public void migrateIfNeeded( File storeDirectory)
             throw new UpgradeNotAllowedByConfigurationException();
         }
 
-        CustomIOConfigValidator.assertCustomIOConfigNotUsed( config, CUSTOM_IO_EXCEPTION_MESSAGE );
-
```

Comment:
Instead of removing this from here and check it later in `StoreMigrator`, does it make sense to instead fail as fast as possible here as soon and `versionToMigrateFrom` is known?
I mean inside the `if ( MigrationStatus.X.isNeededFor( migration status ) )` blocks.